### PR TITLE
Notify only previous Folder & selected Folder in `FolderAdapter`

### PR DESCRIPTION
**Pull request by KevinBoulongne**

### CODE REVIEW

File: app/src/main/java/com/infomaniak/mail/ui/main/menu/FolderAdapter.kt [hunk 1]
```diff
@@ -83,16 +84,24 @@ class FolderAdapter(
...
 
+    @SuppressLint("NotifyDataSetChanged")
     fun setFolders(newFolders: List<Folder>, newCurrentFolderId: String?) {
         currentFolderId = newCurrentFolderId
         folders = newFolders
         notifyDataSetChanged()
     }
 
     fun updateSelectedState(newCurrentFolderId: String) {
+        notifyCurrentItem()
```
Comment (sirambd) [inline app/src/main/java/com/infomaniak/mail/ui/main/menu/FolderAdapter.kt:95]: Why are we notifying twice in a row?
And when do we notify `newCurrentFolderId`?

**Review by sirambd — CHANGES_REQUESTED** [review]

File: app/src/main/java/com/infomaniak/mail/ui/main/menu/FolderAdapter.kt (hunk 1 above)
Comment (KevinBoulongne) [reply app/src/main/java/com/infomaniak/mail/ui/main/menu/FolderAdapter.kt]: This code was wrong, it's been redone.

**Review by JorisBodin — APPROVED** [review]

File: app/src/main/java/com/infomaniak/mail/ui/main/menu/FolderAdapter.kt (hunk 1 above)
Comment (sirambd) [reply app/src/main/java/com/infomaniak/mail/ui/main/menu/FolderAdapter.kt]: Where was it redone?

Comment (KevinBoulongne) [reply app/src/main/java/com/infomaniak/mail/ui/main/menu/FolderAdapter.kt]: In this PR.

**Pull request merged**